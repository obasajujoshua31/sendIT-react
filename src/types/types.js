export const signUpUser = "SIGN_UP_USER";
export const signInUser = "SIGN_IN_USER";
export const createOrder = "CREATE_ORDER";
export const createOrderSuccess = "CREATE_ORDER_SUCCESS";
export const createOrderFailure = "CREATE_ORDER_FAILURE";
export const viewOrders = "VIEW_ORDERS";
export const viewOrder = "VIEW_ORDER";
export const updateParcel = "UPDATE_PARCELS";
export const updateLocation = "UPDATE_PARCEL_LOCATION";
export const cancelOrder = "CANCEL_ORDER";
export const updateParcelStatus = "UPDATE_PARCEL_STATUS";
export const signInSuccess = "SIGN_IN_SUCCESS";
export const signInFailure = "SIGN_IN_FAILURE";
export const signUpSuccess = "SIGN_UP_SUCCESS";
export const signUpFailure = "SIGN_UP_FAILURE";
export const isLoggedOut = "LOGGED_OUT";
export const isAdmin = "IS_ADMIN";
