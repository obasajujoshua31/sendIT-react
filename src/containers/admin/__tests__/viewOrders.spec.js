import React from "react";
import ReactModal from "react-modal";
import {
  AdminViewOrders,
  mapDispatchToProps,
  mapStateToProps
} from "../viewOrders";
import { shallow } from "enzyme";
import sinon from "sinon";
import { signInSuccess, signInFailure } from "../../../types/types";

const props = {
  userParcels: [],
  loadParcels: jest.fn(),
  history: {
    push: jest.fn()
  },
  authStatus: "",
  userParcels: {
    data: {}
  },
  role: false
};

  const mockParcels =  [
    {
      parcel: {
        id: "1",
        status: "PLACED"
      }
    },
    {
      parcel: {
        id: "2",
        status: "DELIVERED"
      }
    },
    {
      parcel: {
        id: "3",
        status: "TRANSITING"
      }
    },
    {
      parcel: {
        id: "4",
        status: "CANCELLED"
      }
    }
  ]

const mockStore = {
  users: {
    singInStatus: signInSuccess
  },
  parcels: {
    parcels: ["Beans"]
  },
  role: false
};
const wrapper = shallow(<AdminViewOrders {...props} />);

describe("Test for User AdminViewOrders", () => {
  it("should render AdminViewOrders component", () => {
    wrapper.setState({
      isLoading: false
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("should dispatch loadparcels", () => {
    const dispatchSpy = sinon.spy();
    const dispatchResult = mapDispatchToProps(dispatchSpy);
    dispatchResult.loadParcels();
    expect(dispatchSpy.callCount).toEqual(1);
  });

  it("should return new object for mapstate to props", () => {
    const stateToPropsResult = mapStateToProps(mockStore);
    expect(stateToPropsResult.userParcels).toEqual(["Beans"]);
    expect(stateToPropsResult.authStatus).toEqual(signInSuccess);
  });

  it("should render view orders", () => {
    wrapper.setState({
      parcels: mockParcels
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push", () => {
    const adminWrapper = shallow(<AdminViewOrders {...props} />);

    adminWrapper.setProps({
      authStatus: signInFailure,
      userParcels: mockParcels
    });
    
    expect(wrapper.instance().props.history.push).toBeCalled();
  });

  it("should call render parcels", () => {
    wrapper.setState({
      parcels: mockParcels,
      showModal: false
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("state modal should be true", () => {
    const viewSpy = jest.spyOn(wrapper.instance(), "onHandleViewOrder");
    viewSpy("1");
    expect(wrapper.instance().state.showModal).toEqual(true);
  });

  it("close Modal should set state to false", () => {
    const viewSpy = jest.spyOn(wrapper.instance(), "closeModal");
    wrapper.setState({
      parcels: mockParcels
    });
    viewSpy();
    expect(wrapper.instance().state.showModal).toEqual(false);
  });
});
