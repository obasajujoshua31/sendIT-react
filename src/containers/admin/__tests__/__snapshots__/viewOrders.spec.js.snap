// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for User AdminViewOrders should call render parcels 1`] = `
<Fragment>
  <div
    className="view-order-container"
  >
    <h1>
      All Available Orders
    </h1>
    <table
      align="center"
      className="table"
    >
      <thead>
        <tr>
          <td>
            Parcel Id
          </td>
          <td>
            Parcel Name
          </td>
          <td>
            Pick Up Location
          </td>
          <td>
            Destination
          </td>
          <td>
            Sent On
          </td>
          <td>
            Status
          </td>
        </tr>
      </thead>
      <tbody>
        <withRouter(Order)
          onHandleViewOrder={[Function]}
          parcel={
            Object {
              "parcel": Object {
                "id": "1",
                "status": "PLACED",
              },
            }
          }
        />
        <withRouter(Order)
          onHandleViewOrder={[Function]}
          parcel={
            Object {
              "parcel": Object {
                "id": "2",
                "status": "DELIVERED",
              },
            }
          }
        />
        <withRouter(Order)
          onHandleViewOrder={[Function]}
          parcel={
            Object {
              "parcel": Object {
                "id": "3",
                "status": "TRANSITING",
              },
            }
          }
        />
        <withRouter(Order)
          onHandleViewOrder={[Function]}
          parcel={
            Object {
              "parcel": Object {
                "id": "4",
                "status": "CANCELLED",
              },
            }
          }
        />
      </tbody>
    </table>
  </div>
</Fragment>
`;

exports[`Test for User AdminViewOrders should render AdminViewOrders component 1`] = `
<Fragment>
  <div
    className="view-order-container"
  >
    <h1>
      All Available Orders
    </h1>
    <table
      align="center"
      className="table"
    >
      <thead>
        <tr>
          <td>
            Parcel Id
          </td>
          <td>
            Parcel Name
          </td>
          <td>
            Pick Up Location
          </td>
          <td>
            Destination
          </td>
          <td>
            Sent On
          </td>
          <td>
            Status
          </td>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</Fragment>
`;

exports[`Test for User AdminViewOrders should render view orders 1`] = `
<Fragment>
  <div
    className="view-order-container"
  >
    <h1>
      All Available Orders
    </h1>
    <table
      align="center"
      className="table"
    >
      <thead>
        <tr>
          <td>
            Parcel Id
          </td>
          <td>
            Parcel Name
          </td>
          <td>
            Pick Up Location
          </td>
          <td>
            Destination
          </td>
          <td>
            Sent On
          </td>
          <td>
            Status
          </td>
        </tr>
      </thead>
      <tbody>
        <withRouter(Order)
          onHandleViewOrder={[Function]}
          parcel={
            Object {
              "parcel": Object {
                "id": "1",
                "status": "PLACED",
              },
            }
          }
        />
        <withRouter(Order)
          onHandleViewOrder={[Function]}
          parcel={
            Object {
              "parcel": Object {
                "id": "2",
                "status": "DELIVERED",
              },
            }
          }
        />
        <withRouter(Order)
          onHandleViewOrder={[Function]}
          parcel={
            Object {
              "parcel": Object {
                "id": "3",
                "status": "TRANSITING",
              },
            }
          }
        />
        <withRouter(Order)
          onHandleViewOrder={[Function]}
          parcel={
            Object {
              "parcel": Object {
                "id": "4",
                "status": "CANCELLED",
              },
            }
          }
        />
      </tbody>
    </table>
  </div>
</Fragment>
`;
